c/tree.c

```diff
@@ -574,51 +585,51 @@ LOCAL void alias_ipv4_networks(MMDBW_tree_s *tree)
     }
 
     MMDBW_network_s ipv4_root_network = resolve_network(tree, "::0.0.0.0", 96);
+    MMDBW_node_s *ipv4_root_node = new_node();
+    MMDBW_record_s ipv4_root_record = {
+        .type       = MMDBW_RECORD_TYPE_FIXED_NODE,
+        .value.node = ipv4_root_node,
+    };
+
+    MMDBW_status status = insert_record_for_network(
+        tree,
+        &ipv4_root_network,
+        &ipv4_root_record,
+        MMDBW_MERGE_STRATEGY_NONE,
+        true
+        );
 
-    // We create an empty record for the aliases to point to initially. We do
-    // not simply use /96, as all of the alias code requires that the node be
-    // aliased to a MMDBW_RECORD_TYPE_NODE node. This is gross and confusing
-    // and should be fixed at some point.
-    remove_network(tree, "::0.0.0.0", 97);
-
-    MMDBW_record_s *ipv4_root_record;
-    MMDBW_status status = find_record_for_network(tree, &ipv4_root_network,
-                                                  false,
-                                                  &return_null,
-                                                  &ipv4_root_record,
-                                                  NULL,
-                                                  NULL);
     free_network(&ipv4_root_network);
-    if (status != MMDBW_SUCCESS) {
-        croak("Unable to find IPv4 root node when setting up aliases");
-    }
 
-    if (MMDBW_RECORD_TYPE_NODE != ipv4_root_record->type) {
-        croak("Unexpected type for IPv4 root record: %s",
-              record_type_name(ipv4_root_record->type));
+    if (status != MMDBW_SUCCESS) {
```

Comment:
Most the code puts the macro first (which I did to protect against `=` vs `==` errors)